cw-orch/src/interchain/interchain_channel_builder.rs

```diff
@@ -0,0 +1,233 @@
+
+    async fn get_hermes() -> IcResult<Hermes> {
+        let docker_helper = DockerHelper::new().await?;
+        docker_helper.get_hermes()
+    }
+
+	pub fn chain_a(&mut self, chain_id: impl Into<String>) -> &mut Self{
+		self.chain_a.chain_id = Some(chain_id.into());
+		self
+	}
+
+	pub fn chain_b(&mut self, chain_id: impl Into<String>) -> &mut Self{
+		self.chain_b.chain_id = Some(chain_id.into());
+		self
+	}
+
+	pub fn add_grpc_a(&mut self, grpc: Grpc) -> &mut Self{
+		self.chain_a.grpc = Some(grpc);
+		self
+	}
+
+	pub fn add_grpc_b(&mut self, grpc: Grpc) -> &mut Self{
+		self.chain_b.grpc = Some(grpc);
+		self
+	}
+
+	pub fn is_local_chain_a(&mut self) -> &mut Self{
+		self.chain_a.is_local_chain = Some(true);
+		self
+	}
+
+	pub fn is_local_chain_b(&mut self) -> &mut Self{
+		self.chain_b.is_local_chain = Some(true);
+		self
+	}
+
+	pub fn port_a(&mut self, port: impl Into<String>) -> &mut Self{
+		self.chain_a.port = Some(port.into());
+		self
+	}
+
+	pub fn port_b(&mut self, port: impl Into<String>) -> &mut Self{
+		self.chain_b.port = Some(port.into());
+		self
+	}
+
+	pub fn connection(&mut self, connection: impl Into<String>) -> &mut Self{
+		self.connection_a = Some(connection.into());
+		self
+	}
+
+	// The channel id id supposed to be the one created on the a side (you can interchange a and b at will to allow for that)
+	// The connection between the 2 chains, The two chains_ids, and the port of chain a should be defined when building a channel object with this method
+	// TODO enforce that with errors ? 
+	pub async fn channel_from(&self, channel_id_a: String) -> Result<InterchainChannel, DaemonError>{
+		
+		// First we need to construct the channels for chain a and chain b
+		let channel_a = self.chain_a.create_grpc_channel().await?;
+		let channel_b = self.chain_b.create_grpc_channel().await?;
+
+		// Then we check that the channel indeed exists
+		let registered_channel = Ibc::new(channel_a.clone()).channel(self.chain_a.port.clone().unwrap(), channel_id_a.clone()).await?;
+		let counterparty = registered_channel.counterparty.unwrap();
+
+		let channel = InterchainChannel::new(
+			registered_channel.connection_hops[0].clone(), // We suppose there is only one connection for this channel
+			InterchainPort{
+			    chain: channel_a,
+			    chain_id: self.chain_a.chain_id.clone().unwrap(),
+			    port: self.chain_a.port.clone().unwrap(),
+			    channel: Some(channel_id_a),
+			},
+			InterchainPort{
+				chain: channel_b,
+				chain_id: self.chain_b.chain_id.clone().unwrap(),
+				port: counterparty.port_id,
+				channel: Some(counterparty.channel_id)
+
+			},
+		);
+
+		Ok(channel)
+	}
+
+	// Create a channel on-chain and relay for it (using the Hermes relayer)
+	// The connection, the two chain_ids and the two ports should be defined when creating this channel
+	// TODO enforce that with errors ? 
+	pub async fn create_channel(&self, channel_version: &str) -> Result<InterchainChannel, DaemonError>{
```

Comment:
This creates a channel and tracks its creation. 
In order to work properly, this needs the full description of the 2 ports
(chain_id, grpc_channel, port).
The connection linking the 2 chains is not mandatory, it can be created from the chain description